### ISSUE TITLE
Issues with multiple submenus with same id

### ISSUE THREAD

**Issue by Lynesth** [php-school/cli-menu#131]
We currently can add multiple submenus with the same "id"
Example:
```php
$menu = (new CliMenuBuilder)
    ->addSubMenu('Options')
        ->addItem(...)
        ->end()
    ->addSubMenu('Options')
        ->addItem(...)
        ->end()
    ->end()
```
Both of them work but only the second one has its parent set and can "Go Back" to the main menu.

We either need to prevent adding a menu with the same ID or, even better, handle it differently (maybe like the `uniqid` that's used with split items ?).

**Comment by AydinHassan** [comment]
`uniqid` sounds good to me!

**Comment by AydinHassan** [comment]
This turns out to be more complicated - If we allow multiple sub menus with the same name then `CliMenuBuilder#getSubMenu()` will not function correctly either.

We can't just add a `uniqid` id on because the user needs to be able to reference it. Not sure what to do here.

**Comment by AydinHassan** [comment]
Should be fixed by #155
